### ISSUE TITLE
summary from interop

### ISSUE THREAD

**Issue by coykMccoyk** [Illumina/interop#279]
Hello,
I’d like to get summary information from our NextSeq runs so I can look at performance across all our runs. I know the SAV software can do it per run, but I’d like to extract the summary table it gives for each of our runs. I’ve looked around Illumina’s github page. I’ve installed interop on our server using pip install interop. I’ve followed the directions here
http://illumina.github.io/interop/summary_table.html
 is “summary” a command within interop?  When I type
 summary My/Run/File
 on the command line, I get command not found.
 I’ve also tried
 interop My/Run/File
interop summary My/Run/File
python interop summary My/Run/File
 Am I supposed to install something else?  I copied the lines from
https://github.com/Illumina/interop/blob/master/docs/src/Tutorial_01_Intro.ipynb
into a single file
run_folder = r"/media/efglserv/seqspace/media/efglserv/seqspace/NextSeq/Runs/211222_NB501106_0401_AHT53TAFX2/"
from interop import py_interop_run_metrics, py_interop_run, py_interop_summary
run_metrics = py_interop_run_metrics.run_metrics()
valid_to_load = py_interop_run.uchar_vector(py_interop_run.MetricCount, 0)
py_interop_run_metrics.list_summary_metrics_to_load(valid_to_load)
run_folder = run_metrics.read(run_folder, valid_to_load)
summary = py_interop_summary.run_summary()
py_interop_summary.summarize_run_metrics(run_metrics, summary)
summary.total_summary().yield_g()
import pandas as pd
columns = ( ('Yield Total (G)', 'yield_g'), ('Projected Yield (G)', 'projected_yield_g'), ('% Aligned', 'percent_aligned'))
rows = [('Non-Indexed Total', summary.nonindex_summary()), ('Total', summary.total_summary())]
d = []
for label, func in columns:
    d.append( (label, pd.Series([getattr(r[1], func)() for r in rows], index=[r[0] for r in rows])))
df = pd.DataFrame.from_items(d)
df
 
called the file Summaryinterops.py, then tried to run in from the command line using
python Summaryinterops.py
 but got a bunch of error messages. I don’t know how to print the error messages to a file. It looks like it doesn’t like how I’ve specified the run folder and it can’t find "/media/efglserv/seqspace/media/efglserv/seqspace/NextSeq/Runs/211222_NB501106_0401_AHT53TAFX2/RunInfo.xml" though I can navigate to that folder and see that file there.
 
Any guidance would be appreciated. Also, did I really need to sign up for a github account to ask this question?

**Comment by ezralanglois** [comment]
The command line tools are stored by OS in:

Apple Mac: https://github.com/Illumina/interop/releases/download/v1.1.23/InterOp-1.1.23-Darwin-AppleClang.tar.gz
Linux: https://github.com/Illumina/interop/releases/download/v1.1.23/InterOp-1.1.23-Linux-GNU.tar.gz
Windows: https://github.com/Illumina/interop/releases/download/v1.1.23/InterOp-1.1.23-Windows-MSVC.zip

I suggest running
```
ls /media/efglserv/seqspace/media/efglserv/seqspace/NextSeq/Runs/211222_NB501106_0401_AHT53TAFX2/RunInfo.xml
```

If this does not work, then you have an issue on your side that I cannot help you with.

**Comment by coykMccoyk** [comment]
I've already installed interop on our server
After downloading from your link above, putting that on our server too I know have a folder 
/home/efglserv/software/InterOp-1.1.23-Linux-Gnu
as well as what I installed using 
pip install interop 
Requirement already satisfied: numpy>=1.16 6 in /home/efglserv/anaconda3/lib/python3.6/site-packages (from interop) (1.18.5)

Still, when I type summary My/Run/Folder I get command not found.  

In looking at the readme in InterOp-1.1.23-Linux-Gnu, there's a Python install section, which I've done, and a From Source section. Do I do that as well?  Do I do the "Building with Interop GCC/Clang" part too?
